cpp/include/cudf/hashing.hpp

```diff
@@ -22,16 +22,23 @@
 
 namespace cudf {
 
-using hash_value_type = uint32_t;  ///< Type of hash value
-
 /**
  * @addtogroup column_hash
  * @{
  * @file
  */
 
 /**
- *  @brief Identifies the hash function to be used
+ * @brief Type of hash value
+ *
+ * @deprecated Since 23.08
+ */
+using hash_value_type = uint32_t;
```

Comment:
This deprecation would affect a number of places. Any suggestion on how we adapt related use cases?

Reply:
That is true. I think this probably should be moved to a different header instead of existing here. 
I will look into this. The same consideration for the enum `hash_id` as well -- I found the enum is used by [`hash_partition()`](https://docs.rapids.ai/api/libcudf/stable/group__reorder__partition.html#ga65fe4ae8d1cbe9db31dbda43ebea0e7a) so it can't go away completely.

Reply:
Why is it deprecated? Are we going to use another more "standard" type?

Reply:
After further thought, I'm going to un-deprecate the `hash_value_type` and `hash_id` since they are used in other places besides the `cudf::hash` function.

Reply:
We’re using `hash_value_type` to mean “the return type of libcudf’s default hash function.” We already hardcode the result type as a member type of the hash function implementation, maybe we can reuse that information?

Reply:
Like `using hash_value_type = cudf::detail::default_hash<int>::result_type`?

Reply:
For hash_partition we still run into some of the same issues with API awkwardness and seeds — we can’t hash-partition by the string result of an MD5 hash, for example. And even if we could, it wouldn’t use a seed. Same for HASH_IDENTITY. We should assess the use cases and determine if we need to accept a hash_id in hash_partition or if that should be deprecated.

Reply:
Another alternative would be to rename/restrict hash_id to be something that defines a subset of hash algorithms usable for this purpose.